### ISSUE TITLE
Requesting HBS page, got 500 HTTP error and "partials not found" message

### ISSUE THREAD

**Issue by fabiobeoni** [forwardemail/supertest#459]
Hello,
when I run supertest request over an handlebars page that includes some partials I get a 500 server error, with message "partial ... not found". Removing the partial from the handlebars template everything works in the test request.

Please note that actually the partial is there, when I invoke the same URL of the request manually (browser) I get a 200 with the partial included in the page.

```
supertest(runningExpressApp)
            .get('/myPageWithPartials')
            .expect('Content-Type', /html/)
            .expect(200)
            .then((res) => {
                //never come here because of error "partial not found"
            });
```
Any suggestion? Thanks, F.

**Comment by rcorral** [comment]
@fabiobeoni did you find a solution?

**Comment by maximls** [comment]
I have the same issue. I think hbs is compiling asynchronously and so it's not ready by the time the expect runs. I wonder if there is any way to delay the expect execution.

**Comment by evisong** [comment]
> I have the same issue. I think hbs is compiling asynchronously and so it's not ready by the time the expect runs. I wonder if there is any way to delay the expect execution.
> 
> UPDATE:
> 
> I got it to work by passing to the request the url of the server that's already running. So request("localhost:3000").get().....
> 
> I'm sure there are better ways to get around it but this one works for my app.

You're right, hbs is reading partials folder asyncly, converting the registration callback to Promise works for me:

```js
async function registerPartialsAsPromise(hbs) {
  await new Promise((resolve, reject) => {
    hbs.registerPartials('/your/partials/dir', err => {
      if (err) {
        return reject(err);
      }
      resolve();
    });
  });
}

it('tests', async () => {
  const expressApp = express();
  await registerPartialsAsPromise(hbs);
  expressApp.set('view engine', 'hbs');
  expressApp.engine('hbs', hbs.__express);
  // ...

  await request(expressApp)
    .get('/')
    .expect(/content_in_partials/);
});
```
